bin/import_public_data.py

```diff
@@ -26,8 +26,29 @@ def save_to_db(db_url, phone_view):
                 print(8 * ' ',r["trip_id"], r["eval_common_trip_id"], r["eval_role"], len(r["evaluation_trip_ranges"]))
                 all_entries = r["location_entries"]  + r["filtered_location_entries"] + r["motion_activity_entries"] + r["transition_entries"]
                 print("Combining %d locations, %d background locations, %d motion activity, %d transitions -> %d" %
-                    (len(r["location_entries"]), len(r["background_location_entries"]), len(r["motion_activity_entries"]), len(r["transition_entries"]), len(all_entries)))
+                    (len(r["location_entries"]), len(r["filtered_location_entries"]), len(r["motion_activity_entries"]), len(r["transition_entries"]), len(all_entries)))
                 eiss.post_entries(db_url, phone_label, all_entries)
+                for tr in r["evaluation_trip_ranges"]:
+                    print(12 * ' ', 30 * "-")
+                    print(12 * ' ',tr["trip_id"], tr.keys())
+                    for sr in tr["evaluation_section_ranges"]:
+                        print(16 * ' ', 30 * "~")
+                        print(16 * ' ',sr["trip_id"], sr.keys())
+                        gt_leg = phone_view.spec_details.get_ground_truth_for_leg(sr["trip_id_base"])
```

Comment:
You should add a check here:
`if gt_leg != None:`
Otherwise it crashes if there is no ground truth available.
This just happened for me when importing unimodal_trip_car_bike_mtv_la

Reply:
hm, but there should always be ground truth available since we generate the timeline data from the spec. Let me try re-running...

Reply:
Ah this is because there are two separate walking sections for each trip - one to walk to the vehicle and one to walk from the vehicle. In the long term, I should ensure that the second walk section is `walk_end` instead of `walk_start`. But now we have collected data with this label and I don't want to relabel, so I will use a minimally more sophisticated check.

```
walk_start 2 ['walk_start', 'walk_start']
walk_start 2 ['walk_start', 'walk_start']
```

Reply:
I am not sure why this is not affecting the `car_scooter_...` timeline, since there are multiple `walk_start`s in it too, including two in the first `freeway` trip.

```
$ grep -r walk_start spec_creation/*.filled.json
spec_creation/car_scooter_brex_san_jose.filled.json:          "id": "walk_start",
spec_creation/car_scooter_brex_san_jose.filled.json:          "id": "walk_start",
spec_creation/car_scooter_brex_san_jose.filled.json:          "id": "walk_start",
spec_creation/train_bus_ebike_mtv_ucb.filled.json:          "id": "walk_start",
spec_creation/unimodal_roundtrip.filled.json:          "id": "walk_start",
spec_creation/unimodal_roundtrip.filled.json:          "id": "walk_start",
spec_creation/unimodal_roundtrip.filled.json:          "id": "walk_start",
spec_creation/unimodal_roundtrip.filled.json:          "id": "walk_start",
```

Reply:
Ah, it's because we iterate through the trips and the second trip has only one `walk_start`

```
walk_start 2 ['walk_start', 'walk_start']
walk_start 1 ['walk_start']
```

Reply:
ok so fix this I can take one of two approaches:
- ensure that all legs in a trip, including the automatically added shim legs are unique going forward. Fix the labels on the existing ground truth segments to match.
    - Note that this is a bit tricky because we may have multiple transfers between similar modes; for e.g. if we have a walk -> bus -> train -> bus -> train -> walk timeline, we will have two bus -> train transfers. We would then need to check for that and make the tripid unique while autofilling the spec.
- in addition to the leg_id, we can pass in the trip_id and the locations for the leg, and return  the leg with the greatest overlap between the spatial ground truth and the actual points.

Reply:
I think that overall, it would be easiest for to ensure that the shim leg ids are unique going forward. This is because I don't want to involve any analysis in figuring out the "ground truth"; that should be as immutable as possible.